docs/multi-homing.md

```diff
@@ -0,0 +1,117 @@
+has put forward a [standard](https://github.com/k8snetworkplumbingwg/multi-net-spec)
+describing how to specify the configurations for additional network interfaces.
+
+There are several delegating plugins or meta-plugins (Multus, Genie) implementing this standard.
+
+After a pod is scheduled on a particular Kubernetes node, kubelet will invoke
+the delegating plugin to prepare the pod for networking. This meta-plugin will
+then invoke the CNI responsible for setting up the pod's default cluster
+network, and afterwards it iterates the list of additional attachments on the
+pod (yes, these plugins **must** be able to access the Kubernetes API...),
+invoking the corresponding delegate CNI implementing the logic to attach the pod
+to that particular network.
+
+## Configuring secondary networks
+To allow pods to have multiple network interfaces, the user must provide the
+configurations specifying how to connect to these networks; these
+configurations are defined in a CRD named `NetworkAttachmentDefinition`.
+
+Below you will find example attachment configurations for each of the current
+topologies OVN-K allows for secondary networks.
+
+**NOTE**: currently, all the secondary networks **only** allow for east/west
+traffic.
+
+### Routed - layer 3 - topology
+This topology is a simplification of the topology for the cluster default
+network - but without egress.
+
+There is a logical switch per node - each with a different subnet - and a
+router interconnecting all the logical switches.
+
+The following net-attach-def configures the attachment to a routed secondary
+network.
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: l3-network
+  namespace: left-one
+spec:
+  config: |2
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "subnets": "10.128.0.0/16/24",
+            "mtu": 1300,
+            "netAttachDefName": "left-one/l3-network"
+    }
+```
+
+**NOTE**
+- the `netAttachDefName` parameter **must** be in <namespace>/<net-attach-def name> format
+- the `subnets` attribute is a comma separated list of subnets.
+- the `subnets` attribute indicates both the subnet across the cluster, and per node.
+  The example above means you have a /16 subnet for the network, but each **node** has
+  a /24 subnet.
+
+### Switched - layer 2 - topology
+This topology interconnects the workloads via a cluster-wide logical switch.
+
+The following net-attach-def configures the attachment to a layer 2 secondary
+network.
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: not-so-fast-network
+  namespace: left-one
+spec:
+  config: |2
+    {
+            "cniVersion": "0.3.1",
+            "name": "not-so-fast-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer2",
+            "subnets": "10.128.0.0/24",
```

Comment:
How does a list of subnets work here? Would be Pods within the network given IPs from random not overlapping subnets?

Reply:
Random ? That word throws me off; they would get IPs from the subnets defined by the user.

Reply:
If I request two subnets here, and then request this NAD from a Pod, what IP (from which subnet) gets assigned to the Pod?

Reply:
one of each ?

Reply:
Aha! It was not clear to me that the Pod will get a IP from each of the listed subnets. My assumption was that they somehow join into one big subnet, that's why I was confused.

Could you mentioned in the docs what's the behavior with multiple subnets listed?

Reply:
```bash
kubectl exec -nns1 tinypod -- ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0@if17: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:f4:00:07 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.244.0.7/24 brd 10.244.0.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::858:aff:fef4:7/64 scope link 
       valid_lft forever preferred_lft forever
3: net1@if18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1300 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:80:00:03 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.128.0.3/24 brd 10.128.0.255 scope global net1
       valid_lft forever preferred_lft forever
    inet6 fe80::858:aff:fe80:3/64 scope link 
       valid_lft forever preferred_lft forever
4: net2@if19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1300 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:80:00:01 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.128.0.1/24 brd 10.128.0.255 scope global net2
       valid_lft forever preferred_lft forever
    inet 192.168.200.1/24 brd 192.168.200.255 scope global net2
       valid_lft forever preferred_lft forever
    inet6 fe80::858:aff:fe80:1/64 scope link 
       valid_lft forever preferred_lft forever

```

Reply:
> Aha! It was not clear to me that the Pod will get a IP from each of the listed subnets. My assumption was that they somehow join into one big subnet, that's why I was confused.
> 
> Could you mentioned in the docs what's the behavior with multiple subnets listed?

Done.

Reply:
thanks I didn't know this either :)